orchagent/portsorch.cpp

```diff
@@ -4363,6 +4452,69 @@ void PortsOrch::doVlanMemberTask(Consumer &consumer)
     }
 }
 
+void PortsOrch::doTransceiverInfoTableTask(Consumer &consumer)
+{
+    /*
+    the idea is to listen to transceiver info table, and also maintain an internal list of plugged modules.
+
+    */
+    SWSS_LOG_ENTER();
+
+    string table_name = consumer.getTableName();
+
+    auto it = consumer.m_toSync.begin();
+    while(it != consumer.m_toSync.end())
+    {
+        auto t = it->second;
+        string alias = kfvKey(t);
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            SWSS_LOG_DEBUG("TRANSCEIVER_INFO table has changed - SET command for port %s", alias.c_str());
+            
+            if (m_pluggedModulesPort.find(alias) == m_pluggedModulesPort.end())
+            {
+                m_pluggedModulesPort[alias] = m_portList[alias];
+                
+                SWSS_LOG_DEBUG("Setting host_tx_signal allow for port %s", alias.c_str());
+                setSaiHostTxSignal(m_pluggedModulesPort[alias], true);
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            SWSS_LOG_DEBUG("TRANSCEIVER_INFO table has changed - DEL command for port %s", alias.c_str());
+            
+            Port p;
+            if (m_pluggedModulesPort.find(alias) != m_pluggedModulesPort.end())
+            {
+                p = m_pluggedModulesPort[alias];
+                m_pluggedModulesPort.erase(alias);
+            }
+            SWSS_LOG_DEBUG("Setting host_tx_signal NOT allow for port %s", alias.c_str());
+            setSaiHostTxSignal(p, false);
+        }
+
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+bool PortsOrch::setSaiHostTxSignal(Port port, bool enable)
+{
+    sai_attribute_t attr;
+    attr.id = SAI_PORT_ATTR_HOST_TX_SIGNAL_ENABLE;
+    attr.value.booldata = enable;
+    sai_status_t status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Could not set port {} attribute {}");
```

Comment:
what is the "{}" used for? Do you mean "%d"? And please add the format argument.

Reply:
Done